src/server/JasmineGraphServer.cpp

```diff
@@ -1319,697 +1129,126 @@ void JasmineGraphServer::copyCentralStoreToAggregateLocation(std::string filePat
     }
+    server_logger.info("Sent: " + fileName);
 
-    server_logger.log("Sent : " + JasmineGraphInstanceProtocol::HANDSHAKE, "info");
-    bzero(data, FED_DATA_LENGTH + 1);
-    read(sockfd, data, FED_DATA_LENGTH);
-    string response = (data);
+    string response = Utils::read_str_trim_wrapper(sockfd, data, FED_DATA_LENGTH);
+    if (response.compare(JasmineGraphInstanceProtocol::SEND_FILE) != 0) {
+        server_logger.error("Incorrect response. Expected: " + JasmineGraphInstanceProtocol::SEND_FILE +
+                            " ; Received: " + response);
+        close(sockfd);
+        return false;
+    }
+    server_logger.info("Received: " + response);
 
-    response = Utils::trim_copy(response, " \f\n\r\t\v");
+    server_logger.info("Sending file " + filePath + " through port " + std::to_string(dataPort));
+    FILE *fp = fopen(filePath.c_str(), "r");
+    if (fp == NULL) {
+        close(sockfd);
+        return false;
+    }
 
-    if (response.compare(JasmineGraphInstanceProtocol::HANDSHAKE_OK) == 0) {
-        server_logger.log("Received : " + JasmineGraphInstanceProtocol::HANDSHAKE_OK, "info");
-        result_wr = write(sockfd, masterIP.c_str(), masterIP.size());
+    while (true) {
+        unsigned char buff[1024] = {0};
+        int nread = fread(buff, 1, 1024, fp);
 
-        if (result_wr < 0) {
-            server_logger.log("Error writing to socket", "error");
+        /* If read was success, send data. */
+        if (nread > 0) {
+            write(sockfd, buff, nread);
         }
 
-        server_logger.log("Sent : " + masterIP, "info");
-        bzero(data, FED_DATA_LENGTH + 1);
-        read(sockfd, data, FED_DATA_LENGTH);
-        response = (data);
-
-        if (response.compare(JasmineGraphInstanceProtocol::HOST_OK) == 0) {
-            server_logger.log("Received : " + JasmineGraphInstanceProtocol::HOST_OK, "info");
-        } else {
-            server_logger.log("Received : " + response, "error");
+        if (nread < 1024) {
+            if (feof(fp)) printf("End of file\n");
+            if (ferror(fp)) printf("Error reading\n");
+            break;
         }
+    }
 
-        result_wr = write(sockfd, JasmineGraphInstanceProtocol::UPLOAD_RDF_ATTRIBUTES.c_str(),
-                          JasmineGraphInstanceProtocol::UPLOAD_RDF_ATTRIBUTES.size());
+    fclose(fp);
+    close(sockfd);
+    return true;
+}
 
-        if (result_wr < 0) {
-            server_logger.log("Error writing to socket", "error");
-        }
+static void copyArtifactsToWorkers(const std::string &workerPath, const std::string &artifactLocation,
+                                   const std::string &remoteWorker) {
+    if (artifactLocation.empty() || artifactLocation.find_first_not_of(' ') == artifactLocation.npos) {
+        server_logger.error("Received `" + artifactLocation + "` for `artifactLocation` value!");
+        throw std::invalid_argument("Received empty string for `artifactLocation` value!");
```

Comment:
I there point of throwing an exception here? I do not see a use case for JasmineGraph server class as a component. Therefore, logging only would be sufficient.

Reply:
Replaced with `return` in next PR

Reply:
Ok